graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java

```diff
@@ -131,7 +131,7 @@ public void createEvents(EventFactory eventFactory, EventProcessorParameters pro
             if (config.series().isEmpty()) {
                 filterSearch(eventFactory, parameters, eventsConsumer);
             } else {
-                aggregationSearchUtils.aggregatedSearch(eventFactory, parameters, eventsConsumer);
+                aggregationSearchUtils.aggregatedSearch(eventFactory, parameters, eventsConsumer, null);
```

Comment:
Please add an overloaded `#aggregatedSearch` method instead of adding a nullable parameter. The method that doesn't have the `eventDecorator` parameter can just pass `Function.identity()` to the overloaded function.

`.aggregatedSearch(eventFactory, parameters, eventConsumer, Function.identity())`

Reply:
Ah, right, I missed that, sorry. If the decorator does in-place modifications and doesn't return anything, we can use a `Callable<Event>` instead of a `Function<Event, Void>` and an empty decorator function, as you suggested.

Reply:
Not super familiar with `Callable` but it looks like a `Callable<Event>` would be a task that returns an `Event` and doesn't take any parameters. So if we wanted it to operate on the `Event` then we'd need access to it when defining the `Callable`, but that would not be the case when calling this from the code in the enterprise PR.

Reply:
Of course I got it wrong again, I meant `Consumer<Event>`. Sorry...

Reply:
I assumed there was some other interface that worked but couldn't find it quickly enough, thank you! Will rework now!